### DISCUSSION TITLE
How to add custom combos in Miryoku ZMK?

### DISCUSSION THREAD

**Discussion by mplusp** [manna-harbour/miryoku#321] [Q&A] [answered]
Hi, first of all: I'm loving Miryoku! Thanks for all your hard work. 

I'm using a Chocofi, using the GitHub Actions build for the Corne. For alphas I'm using the default. I'd like to add a combo to the base layer to switch to another custom layer. I don't have it completely planned out, but I'm already failing at defining a simple combo at the base layer for P+B.

 I found [this thread](https://github.com/manna-harbour/miryoku/discussions/233) but still failed to get this implemented.

Any help would be appreciated. Thanks for your time.

**Comment by manna-harbour** [comment, marked as answer]
To switch to an alternative base layer, you can substitute the Extra or Tap layers, no combos needed. See https://github.com/manna-harbour/miryoku/tree/master/docs/reference#additional-features and https://github.com/manna-harbour/miryoku/discussions/85.

If you want custom combos anyway, see https://github.com/manna-harbour/miryoku/issues/94 and https://github.com/manna-harbour/miryoku/discussions/233.

If you don't want to deal with the `cpp` macros in https://github.com/manna-harbour/miryoku/discussions/233, you can just add combos according to https://zmk.dev/docs/features/combos, by appending to https://github.com/manna-harbour/miryoku_zmk/blob/master/config/corne.keymap, binding e.g. `&to U_EXTRA`, and using the native corne 3x6+3 key position indices.
